viadot/sources/eurostat.py

```diff
@@ -0,0 +1,256 @@
+                )
+        except:
+            self.logger.error(f"Failed to make parameters validation.")
+
+    def eurostat_dictionary_to_df(self, *signals: list) -> pd.DataFrame:
+        """Function for creating DataFrame from json pulled from Eurostat
+
+        Returns:
+            pd.DataFrame: with 4 coumns: index, geo, time, indicator
+        """
+
+        class T_SIGNAL:
+            signal_keys_list: list
+            signal_index_list: list
+            signal_label_list: list
+            signal_name: str
+
+        # Dataframe creation
+        columns0 = signals[0].copy()
+        columns0.append("indicator")
+        df = pd.DataFrame(columns=columns0)
+        indicator_list = []
+        index_list = []
+        signal_lists = []
+
+        # get the dictionary from the inputs
+        eurostat_dictionary = signals[-1]
+
+        for signal in signals[0]:
+            signal_struct = T_SIGNAL()
+            signal_struct.signal_name = signal
+            signal_struct.signal_keys_list = list(
+                eurostat_dictionary["dimension"][signal]["category"]["index"].keys()
+            )
+            signal_struct.signal_index_list = list(
+                eurostat_dictionary["dimension"][signal]["category"]["index"].values()
+            )
+            signal_label_dict = eurostat_dictionary["dimension"][signal]["category"][
+                "label"
+            ]
+            signal_struct.signal_label_list = [
+                signal_label_dict[i] for i in signal_struct.signal_keys_list
+            ]
+            signal_lists.append(signal_struct)
+
+        col_signal_temp = []
+        row_signal_temp = []
+        for row_index, row_label in zip(
+            signal_lists[0].signal_index_list, signal_lists[0].signal_label_list
+        ):  # rows
+            for col_index, col_label in zip(
+                signal_lists[1].signal_index_list, signal_lists[1].signal_label_list
+            ):  # cols
+                index = str(
+                    col_index + row_index * len(signal_lists[1].signal_label_list)
+                )
+                if index in eurostat_dictionary["value"].keys():
+                    index_list.append(index)
+                    col_signal_temp.append(col_label)
+                    row_signal_temp.append(row_label)
+
+        indicator_list = [eurostat_dictionary["value"][i] for i in index_list]
+        df.indicator = indicator_list
+        df[signal_lists[1].signal_name] = col_signal_temp
+        df[signal_lists[0].signal_name] = row_signal_temp
+
+        return df
+
+    def get_data_frame_from_response(self) -> pd.DataFrame:
+        """Function responsible for getting response, creating DataFrame using method 'eurostat_dictionary_to_df'
+           with validation of provided parameters and their codes if needed.
+
+        Returns:
+            pd.DataFrame: final DataFrame or raise prefect.logger.error, if issues occur.
```

Comment:
Start with a capital letter.
You should raise/catch a specific error and add the information in the docstring under `Raises` section.(sources/bigquery for reference).

Reply:
done.